File: src/main/java/com/aoranzhang/ezrentback/data/entity/RoomType.java
package com.aoranzhang.ezrentback.data.entity;

import com.fasterxml.jackson.annotation.JsonValue;

public enum RoomType {
    BEDROOM("bedroom"),
    WASHROOM("washroom"),
    KITCHEN("kitchen"),
    MASTER("master bedroom");

    public static RoomType privateRooms[] = {BEDROOM, MASTER};
    public static RoomType publicRooms[] = {WASHROOM, KITCHEN};

    private String type;

    RoomType(String type) {
        this.type = type;
    }

    @JsonValue
    public String type() {
        return type;
    }

    @Override
    public String toString() {
        return type;
    }

    public String getName() {
        return type;
    }
}

File: src/main/kotlin/com/aoranzhang/ezrentback/data/entity/RoomType.kt
package com.aoranzhang.ezrentback.data.entity

import com.fasterxml.jackson.annotation.JsonValue

enum class RoomType {
    BEDROOM,
    WASHROOM,
    KITCHEN,
    MASTER;

    @JsonValue
    fun type(): String {
        return this.name
    }

    override fun toString(): String {
        return this.name
    }

    companion object {
        var privateRooms = arrayOf(BEDROOM, MASTER)
        var publicRooms = arrayOf(WASHROOM, KITCHEN)
    }
}
